src/main/java/uk/gov/cslearning/catalogue/api/ReportController.java

```diff
@@ -20,10 +22,17 @@
     private static final PageRequest MAX_PAGEABLE = PageRequest.of(0, 10000);
     private final EventService eventService;
     private final ModuleService moduleService;
+    private final CourseService courseService;
 
-    public ReportController(EventService eventService, ModuleService moduleService) {
+    public ReportController(EventService eventService, ModuleService moduleService, CourseService courseService) {
         this.eventService = eventService;
         this.moduleService = moduleService;
+        this.courseService = courseService;
+    }
+
+    @GetMapping("/mandatory-courses")
+    public ResponseEntity<Map<String, CourseDto>> getPublishedAndArchivedMandatoryCourses() {
+        return ResponseEntity.ok(courseService.getPublishedAndArchivedMandatoryCourses());
```

Comment:
What if the service throws error/exception?

Reply:
It will behave as per the existing error handling framework (if exists) and a standard error page will be shown to the user.

Reply:
Ok. We need a wider error handling for this application and we address this part of that.

Reply:
I raised this as users are also getting inconsistent error pages/messages; so this will be addressed as part of that ticket when assigned.

https://cshrdigitalandanalysis.atlassian.net/browse/LC-354